app/sdk/controllers/ConversionController.java

```
@@ -48,20 +58,21 @@ public Result getConversionConfiguration(String endpointName) {
         return ok(jsonNode);
     }
 
-
-    protected DataSetItem dataSetItemFromRequest(ServiceConfiguration configuration, Http.Request request) {
+    protected DataSetItem dataSetItemFromRequest(ServiceConfiguration configuration, Http.Request request, String fieldName) {
         ObjectNode node = (ObjectNode) request.body().asJson();
         DataSet newDataSet = new DataSet(configuration.getAttributes());
-        ObjectNode sourceItem = (ObjectNode) node.findValue("dataSetItem");
-        return dataSetItemForJSON(sourceItem, newDataSet, false, new HashMap<>());
+        JsonNode sourceItemNode = node.findValue(fieldName);
+        if (sourceItemNode == null || sourceItemNode instanceof NullNode) {
+            return null;
```

I'm not a big fan of returning null, should you throw an exception instead?